transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/wizards/Hl7Page.java

```diff
@@ -0,0 +1,570 @@
+                                model.setSourceType(ModelType.JSON);
+                            }
+                            model.setSourceFilePath(path);
+                        } else {
+                            if (schema) {
+                                model.setTargetType(ModelType.JSON_SCHEMA);
+                            } else {
+                                model.setTargetType(ModelType.JSON);
+                            }
+                            model.setTargetFilePath(path);
+                        }
+                        hl7FileText.setText(path);
+                        hl7PreviewText.setText(jsonText);
+                        notifyControl(hl7FileText, SWT.Modify);
+                    } catch (final Exception e) {
+                        Activator.error(e);
+                    }
+                }
+            }
+        });
+
+        final Group group2 = new Group(_page, SWT.SHADOW_ETCHED_IN);
+        group2.setText("JSON Structure Preview");
+        group2.setLayout(new FillLayout());
+        group2.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 3, 3));
+
+        hl7PreviewText = new Text(group2, SWT.V_SCROLL | SWT.READ_ONLY | SWT.H_SCROLL );
+        hl7PreviewText.setBackground(_page.getBackground());
+
+        bindControls();
+        validatePage();
+    }
+
+    private String getJsonText(IFile jsonFile) {
+        if (jsonFile != null && jsonFile.exists()) {
+            try (final InputStream istream = jsonFile.getContents()) {
+                final StringBuffer buffer = new StringBuffer();
+                try (final BufferedReader in = new BufferedReader(new InputStreamReader(istream))) {
+                    String inputLine;
+                    while ((inputLine = in.readLine()) != null) {
+                        buffer.append(inputLine + "\n");
+                    }
+                }
+                return buffer.toString();
+            } catch (final CoreException e1) {
+                e1.printStackTrace();
+            } catch (final IOException e1) {
+                e1.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean isSourcePage() {
+        return isSource;
+    }
+
+    @Override
+    public boolean isTargetPage() {
+        return !isSource;
+    }
+
+    boolean fileIsEmpty(final String path) {
+        try {
+            IFile testIFile = model.getProject().getFile(path);
+            if (testIFile.exists()) {
+                File testFile = testIFile.getRawLocation().makeAbsolute().toFile();
+                if (testFile.length() == 0) {
+                    return true;
+                }
+                String jsonText = getJsonText(testIFile);
+                if (jsonText != null && jsonText.trim().isEmpty()) {
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        } catch (Exception e) {
+            // ignore
```

Comment:
ignored and unlogged exceptions will make our lives harder when looking for bugs